ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -508,6 +519,122 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
             metrics.deleteBlobTotalTimeInMs, null, null, totalTimeSpent));
   }
 
+  /**
+   * Handles a {@link BatchDeleteRequest} and sends out the response via the {@link RequestResponseChannel}.
+   * @param request the {@link NetworkRequest} that contains the {@link BatchDeleteRequest}.
+   * @throws IOException if there is an I/O error reading from the {@link NetworkRequest}.
+   * @throws InterruptedException if the operation is interrupted.
+   */
+  @Override
+  public void handleBatchDeleteRequest(NetworkRequest request) throws IOException, InterruptedException {
+    logger.info("Batch delete request received.");
+    BatchDeleteRequest batchDeleteRequest;
+    if (request instanceof LocalChannelRequest) {
+      // This is a case where handleDeleteRequest is called when frontends are talking to Azure. In this case, this method
+      // is called by request handler threads running within the frontend router itself. So, the request can be directly
+      // referenced as java objects without any need for deserialization.
+      batchDeleteRequest = (BatchDeleteRequest) ((LocalChannelRequest) request).getRequestInfo().getRequest();
+    } else {
+      batchDeleteRequest = BatchDeleteRequest.readFrom(new DataInputStream(request.getInputStream()), clusterMap);
+    }
+    long requestQueueTime = SystemTime.getInstance().milliseconds() - request.getStartTimeInMs();
+    long totalTimeSpent = requestQueueTime;
+    long startTime = SystemTime.getInstance().milliseconds();
+    boolean operationSuccessful = true;
+    ServerErrorCode operationServerErrorCode = ServerErrorCode.No_Error;
+    List<BatchDeletePartitionResponseInfo> partitionResponseInfoList = new ArrayList<>();
+    BatchDeleteResponse response = new BatchDeleteResponse(batchDeleteRequest.getCorrelationId(), batchDeleteRequest.getClientId(), new ArrayList<>(), ServerErrorCode.No_Error);
+    for (BatchDeletePartitionRequestInfo batchDeletePartitionRequestInfo : batchDeleteRequest.getPartitionRequestInfoList()) {
+      BatchDeletePartitionResponseInfo partitionResponseInfo = null;
+      try {
+        List<StoreKey> convertedStoreKeys = getConvertedStoreKeys(batchDeletePartitionRequestInfo.getBlobIds());
+        ServerErrorCode error =
+            validateRequest(batchDeletePartitionRequestInfo.getPartition(), RequestOrResponseType.BatchDeleteRequest,
```

Comment:
This method is name confusing. It is validating partition and whether PUT request is going to READ only partition. Also, FYI @piyujai it doesn't use `skipPartitionAndDiskAvailableCheck` in case you are relying on this check for disk and partition availability